src/storage-controller/src/collection_mgmt.rs

```diff
@@ -201,8 +201,12 @@ where
     ///
     /// Update the `desired` state of a differential collection using
     /// [Self::differential_write].
-    pub(super) fn register_differential_collection<'a, R>(&self, id: GlobalId, read_handle_fn: R)
-    where
+    pub(super) fn register_differential_collection<'a, R>(
+        &self,
+        id: GlobalId,
+        read_handle_fn: R,
+        migrated_storage_collections: &BTreeSet<GlobalId>,
```

Comment:
I would prefer if we could not let the idea of migrated storage collections seep into this one as well. Maybe we can add a `force_read_only: Option<boolean>`, and we compute that at the call-sites?

And you could put the logic that now is in two places in a helper method in the controller, and therefore also only have the explanatory comment there? Something like `is_force_read_only`, but I'm bad with naming... 😅

Reply:
I don't think I understood your suggestion 100%, but I just pushed something, can you take another look? I ended up inverting the boolean.